docs/references/annotations.md

```diff
@@ -90,9 +110,25 @@ metadata:
 will target Kong Ingress controller, forcing the GCE controller
 to ignore it.
 
-> Deploying multiple ingress controller and not specifying the
-annotation will cause both controllers fighting to satisfy the Ingress
-and will lead to unknown behaviour.
+The following resources _require_ this annotation by default:
+
+- Ingress
+- KongConsumer
+- TCPIngress
+- KongClusterPlugin
+- Secret resources with the `ca-cert` label 
```

Comment:
I chose to write this as I did because it's brief--we require this annotation or similar only on a small set (5 total) of resources. An exhaustive list would not be brief:

```
kubectl api-resources | wc -l
116
```

For the vast majority, it simply does nothing, and there are no functional consequences to accidentally adding it elsewhere.

User confusion is a _possible_ concern, but not one I see enough evidence for to warrant an exhaustive list: anecdotally, existing user confusion is more centered around resources that require the resource, where the user wasn't aware that was the case or didn't understand the nuances of the "empty allowed, but only if you didn't override the default controller class, and also not during updates for some things because bugs" legacy behavior. I have not seen questions around annotating a resource with a non-matching class and still seeing the controller load it; if we do start receiving those in the future, we can revisit this. Leaving as-is for now.

Reply:
As discussed elsewhere, KIC operates using ~12 object kinds (judging by informers registered [in the code](https://github.com/Kong/kubernetes-ingress-controller/blob/39bfb2b8903f082ac95f927c31b5b4f918393264/cli/ingress-controller/main.go#L292)), not 116, so it should be sufficient only to cover those 12.

> there are no functional consequences to accidentally adding it elsewhere.

But there are consequences to making an effectively non-functional configuration change by mistake, and one expectation I have (and, I believe, we should have in this case - we can discuss if that seems unreasonable) from documentation is to prevent such mistakes. This one seems very simple to prevent.

> I chose to write this as I did because it's brief

I do not see how a table with 12 entries is not brief, keeping in mind the complexity of the subject. Table is not a format I'll strongly enforce here. My requirement is that we have an authoritative source which answers the question whether that specific kind requires an ingress class or not (with an equally strong "yes" and "no"), and how to set it.

Reply:
Noted, left as-is.

Reply:
I request that we _do_ (in whatever form, if the table suggested above does not work) provide an answer to the question _Does resource X require an ingress class annotation?_ in our docs, especially given the fact, that there are at least 4 different ways for Kong to decide whether to include a resource in config or not (the annotation, the ingressv1 way, the knative way, the dependent resource way)